src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -120,9 +121,12 @@ public StreamingResponseBody streamEvents(
                 }
 
                 // check that offsets are not out of bounds and partitions exist
-                if (cursors != null && !topicRepository.areCursorsValid(topic, cursors)) {
-                    writeProblemResponse(response, outputStream, PRECONDITION_FAILED, "cursors are not valid");
-                    return;
+                if (cursors != null) {
+                    final Optional<String> errorMessage = topicRepository.validateCursors(topic, cursors);
```

Comment:
@antban I did some changes. I did not refactored other code that uses cursos. I just changed the validation aspect of it. Tried to move things around in order to get a better design as by your suggestions.